build/MUX-CI.yml

```diff
@@ -23,90 +23,31 @@ jobs:
-    inputs:
-      restoreSolution: MUXControls.sln
-      feedsToUse: config
+  - template: AzurePipelinesTemplates\MUX-BuildProject-Steps.yml
+    parameters:
+      solutionPath: MUXControls.sln
       nugetConfigPath: nuget.config
-
-  - task: VSBuild@1
-    displayName: 'Build solution MUXControls.sln'
-    inputs:
-      solution: MUXControls.sln
-      vsVersion: 15.0
-      platform: '$(buildPlatform)'
-      configuration: '$(buildConfiguration)'
-      msbuildArgs: '/p:AppxPackageDir="$(appxPackageDir)" /p:AppxBundle=Never /p:AppxSymbolPackageEnabled=false'
-
-  - task: powershell@2
-    displayName: 'Copy files to staging dir'
-    inputs:
-      targetType: filePath
-      filePath: build\CopyFilesToStagingDir.ps1
-      arguments: -BuildOutputDir '$(buildOutputDir)' -PublishDir '$(Build.ArtifactStagingDirectory)' -Platform '$(buildPlatform)' -Configuration '$(buildConfiguration)'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish artifact: drop'
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-      artifactName: 'drop'
-
-
-- job: CreateNuGet
-  dependsOn:
-    - Build
-  variables:
-    nupkgdir: '$(build.artifactStagingDirectory)'
-    buildOutputDir: '$(Build.SourcesDirectory)\BuildOutput'
-  steps:
-  - task: DownloadBuildArtifacts@0 
-    inputs: 
-      artifactName: drop 
-      downloadPath: '$(buildOutputDir)'
-
-  - task: NuGetToolInstaller@0
-    displayName: 'Use NuGet 4.9.1'
-    inputs:
-      versionSpec: 4.9.1
-
-    # Extract the build revision number from Build.BuildNumber. This is needed to pass to build-nupkg
-  - powershell: |
-      $revision = $env:BUILD_BUILDNUMBER.substring($env:BUILD_BUILDNUMBER.length - 3, 3)
-      Write-Host "##vso[task.setvariable variable=buildrevision]$revision"
-    displayName: 'Get build revision number'
-
-  - task: powershell@2
-    displayName: 'build-nupkg.ps1'
-    inputs:
-      targetType: filePath
-      filePath: build\NuSpecs\build-nupkg.ps1
-      arguments: -BuildOutput '$(buildOutputDir)\drop' -OutputDir '$(nupkgdir)' -prereleaseversion prerelease -Subversion '$(buildrevision)' -SkipFrameworkPackage
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish artifact: nupkg'
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-      artifactName: 'drop'
+      copyFilesToStaging: true
+
+- template: AzurePipelinesTemplates\MUX-CreateNugetPackage-Job.yml
+  parameters:
+    dependsOn: Build
+  
+- template: AzurePipelinesTemplates\MUX-NugetReleaseTest-Job.yml
+  parameters:
+    dependsOn: CreateNugetPackage
+    matrix: 
+      Debug_x86:
+        buildPlatform: 'x86'
+        buildConfiguration: 'Debug'
+      Debug_x64:
+        buildPlatform: 'x64'
+        buildConfiguration: 'Debug'
+      Release_x64:
+        buildPlatform: 'x64'
+        buildConfiguration: 'Release'
+      Release_arm:
+        buildPlatform: 'arm'
+        buildConfiguration: 'Release'
```

Comment:
Can we reuse the matix which defined in line 9?

Reply:
No, it's (intentionally) a different matrix. We don't need to build all release configs here and they're slower to build.